pkgs/unified_analytics/lib/src/event.dart

```diff
@@ -4,11 +4,15 @@
 
 import 'dart:convert';
 
+import 'package:collection/collection.dart';
+
 import 'enums.dart';
 
 final class Event {
   final DashEvent eventName;
   final Map<String, Object?> eventData;
+  final DeepCollectionEquality _deepCollectionEquality =
+      const DeepCollectionEquality();
```

Comment:
do we need a new instance of this for every class? I think we can make this a top level private var in the library that we can re-use for each Event instance.